crates/wasi/src/p3/filesystem/host.rs

```diff
@@ -848,23 +793,22 @@ impl types::HostDescriptorWithStore for WasiFilesystem {
             let dir = Arc::clone(dir.as_dir());
             let (result_tx, result_rx) = oneshot::channel();
             let stream = if allow_blocking_current_thread {
-                StreamReader::new(
-                    instance,
-                    &mut store,
-                    BlockingDirectoryStreamProducer {
-                        dir,
-                        result: Some(result_tx),
-                    },
-                )
+                match dir.entries() {
```

Comment:
In p2 this operation is performed in a separate thread (if runtime is configured to do so), so I assumed this may run for a while and avoided calling it from within `read_directory`. Since `read_directory` is *not* marked `async`, I think the expectation is that it should return as soon as possible, hence why IMO `entries()` should only be called *after* `stream.read`.

Of course, if cost of calling `entries` is negligible that does not matter. Are we sure this function will not block for long?